lib/streamlit/runtime/caching/memo_decorator.py

```diff
@@ -116,17 +118,17 @@ class MemoCaches(CacheStatsProvider):
 
     def __init__(self):
         self._caches_lock = threading.Lock()
-        self._function_caches: Dict[str, "MemoCache"] = {}
+        self._function_caches: dict[str, MemoCache] = {}
 
     def get_cache(
         self,
         key: str,
         persist: Optional[str],
-        max_entries: Optional[Union[int, float]],
-        ttl: Optional[Union[int, float]],
+        max_entries: int | float | None,
+        ttl: int | float | None,
```

Comment:
What do people think about using the new union syntax for optional types, either in general or specifically when already part of a union?

I'm not sure myself

Reply:
```python
ttl: Optional[int | float]
vs
ttl: int | float | None
```
FWIW (probably not much), I think I have a slight preference for the latter